src/lib/zod/index.ts

```diff
@@ -16,3 +16,19 @@ export const createCircleSchemaInput = z
       !(data.protocol_name && data.protocol_id),
     'Either Protocol name should be filled in or a Protocol should be selected.'
   );
+
+// this shape mirrors the shape of the original rest endpoint
+// it might be preferable to fold circle_id into the object
+export const createUserSchemaInput = z.object({
+  circle_id: z.number(),
+  object: z.object({
+    name: z.string().min(3).max(255),
+    address: zEthAddressOnly,
+    non_giver: z.boolean().optional(),
+    starting_tokens: z.number().optional(),
+    give_token_remaining: z.number().optional(),
+    fixed_non_receiver: z.boolean().optional(),
+    non_receiver: z.boolean().optional(),
+    role: z.number().min(0).max(1).optional(),
+  }),
+});
```

Comment:
We should also set this to [strict](https://github.com/colinhacks/zod#strict) so Zod invalidates it if it contains any extra fields we don't expect the user to be able to set / modify.